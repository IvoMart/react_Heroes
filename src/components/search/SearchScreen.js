import React, {useMemo} from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import queryString from 'query-string';
import { HeroeCard } from '../../data/HeroeCard';
import { getHeroeByName } from '../../helpers/getHeroeByName';
import { useForm } from '../hooks/useForm';

export const SearchScreen = () => {

  const navigate = useNavigate();
  const {search} = useLocation();

  const {q = ""} =  queryString.parse(search);

  
  const [values, handleInputChange] = useForm({
    searchText: q,
  });
  
  const { searchText } = values;
  //@ts-ignore
  const heroeSearch = useMemo(() => getHeroeByName(q), [q]);
  
  const handlerSearch = (e) => {
    e.preventDefault();
    navigate(`?q=${searchText}`);
  }
  
  return (
    <>
      <h1>Buscar Heroe</h1>
      <hr />
      <div className="row">
        <div className="col-6">
          <h3>Buscar</h3>
          <form onSubmit={handlerSearch}>
            <input type="text" placeholder='Heroe .. ' className='form-control' name='searchText' 
            //@ts-ignore
            onChange={handleInputChange} value={searchText} />

            <button type="submit" className='btn btn-info mt-3' onClick={handlerSearch}>Buscar...</button>
          </form>
        </div>
        <div className="col-4 animate__animated animate__fadeInUp">
          <h4>Resultado</h4>
          {
            (q === '') ? <div className="alert alert-info animate__animated animate__fadeInDown">Buscar un Heroe</div>
                      : (heroeSearch.length === 0) && <div className="alert alert-danger animate__animated animate__fadeInDown">No hay registro para {q}</div>
          }
          {
            heroeSearch.map((heroe)=>(
              <HeroeCard key={heroe.id} {...heroe }/>
            ))
          }
        </div>
      </div>
    </>
  );
};