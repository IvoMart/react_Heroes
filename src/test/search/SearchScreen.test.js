import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { SearchScreen } from '../../components/search/SearchScreen';

describe('Pruebas en <SearchScreen />', () => {
  test('Debe mostrarse con valores por defecto', () => {
    
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search']}>
        <SearchScreen/>
      </MemoryRouter>
    )
    
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('.alert-info').text().trim()).toBe('Buscar un Heroe');
  });
  
});
