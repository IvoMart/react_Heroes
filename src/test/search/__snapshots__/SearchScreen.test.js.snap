// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen /> Debe mostrarse con valores por defecto 1`] = `
Array [
  <h1>
    Buscar Heroe
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h3>
        Buscar
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Heroe .. "
          type="text"
          value=""
        />
        <button
          className="btn btn-info mt-3"
          onClick={[Function]}
          type="submit"
        >
          Buscar...
        </button>
      </form>
    </div>
    <div
      className="col-4 animate__animated animate__fadeInUp"
    >
      <h4>
        Resultado
      </h4>
      <div
        className="alert alert-info animate__animated animate__fadeInDown"
      >
        Buscar un Heroe
      </div>
    </div>
  </div>,
]
`;
